packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,186 @@
+      const eventTime = new Date();
+
+      const values = [eventTime, userId, userEmail, organizationId, eventType, eventMetadata];
+
+      await this.clickHouse.insert({
+        query: sql`
+        INSERT INTO audit_log
+        (event_time, user_id, user_email, organization_id, event_action, metadata)
+        FORMAT CSV`,
+        data: [values],
+        timeout: 5000,
+        queryId: 'create-audit-log',
+      });
+    } catch (error) {
+      this.logger.error('Failed to create audit log event', error);
+      Sentry.captureException(error, {
+        extra: {
+          event,
+        },
+      });
+    }
+  }
+
+  async getPaginatedAuditLogs(
+    props: AuditLogsArgs,
+  ): Promise<{ total: number; data: AuditLogType[] }> {
+    this.logger.info(
+      'Getting paginated audit logs (organization=%s, filter=%o, pagination=%o)',
+      props.selector.organization,
+      props.filter,
+      props.pagination,
+    );
+
+    if (!props.selector.organization) {
+      throw new Error('Organization ID is required');
+    }
+
+    const sqlLimit = sql.raw(props.pagination.limit.toString());
+    const sqlOffset = sql.raw(props.pagination.offset.toString());
+
+    const where: SqlValue[] = [];
+    where.push(sql`organization_id = ${props.selector.organization}`);
+
+    if (props.filter?.userId) {
+      where.push(sql`user_id = ${props.filter.userId}`);
+    }
+
+    if (props.filter?.startDate && props.filter?.endDate) {
+      const from = this.formatToClickhouseDateTime(props.filter.startDate.toISOString());
+      const to = this.formatToClickhouseDateTime(props.filter.endDate.toISOString());
+      where.push(sql`event_time >= ${from} AND event_time <= ${to}`);
+    }
+
+    const whereClause = where.length > 0 ? sql`WHERE ${sql.join(where, ' AND ')}` : sql``;
+
+    const result = await this.clickHouse.query({
+      query: sql`
+        SELECT event_time, user_id, user_email, organization_id, event_action, metadata
+        FROM audit_log
+        ${whereClause}
+        ORDER BY event_time DESC
+        LIMIT ${sqlLimit}
+        OFFSET ${sqlOffset}
+      `,
+      queryId: 'get-audit-logs',
+      timeout: 5000,
+    });
+
+    const totalResult = await this.clickHouse.query({
+      query: sql`
+        SELECT COUNT(*)
+        FROM audit_log
+        ${whereClause}
+      `,
+      queryId: 'get-audit-logs-total',
+      timeout: 5000,
+    });
+
+    return {
+      total: totalResult.rows,
+      data: AuditLogClickhouseArrayModel.parse(result.data),
+    };
+  }
+
+  async resolveRecordAuditLog(event: AuditLogType, injector: Injector) {
+    const currentOrganization = await injector.get(OrganizationManager).getOrganization({
+      organizationId: event.organization_id,
+    });
+    return {
+      userEmail: event.user_email,
+      userId: event.user_id,
+      organizationId: event.organization_id,
+      user: event.metadata.user,
+      organization: currentOrganization,
+    };
+  }
```

Comment:
This method is used in `<event>AuditLog.record` resolver. If we render 100 logs, we make 100 DB calls.